zsh-ai-commands.zsh

```diff
@@ -3,50 +3,97 @@
+      {
+        "role": "user",
+        "content": "Description of what the command should do:\n'''"\nlist files, sort by descending size\n"'''\nGive me the appropriate command."
+      },
+      {
+        "role": "assistant",
+        "content": "ls -lS ## -l long listing -S sort by file size -r reverse order"
+      },
+      {
+        "role": "user",
+        "content": "'$ZSH_AI_COMMANDS_GPT_USER'"
+      }
+    ]
+  }'
+  else
+    ZSH_AI_COMMANDS_GPT_SYSTEM="You only answer 1 appropriate shell one liner that does what the user asks for. The command has to work with the $(basename $SHELL) terminal. Don't wrap your answer in anything, dont acknowledge those rules, don't format your answer. Just reply the plaintext command."
+  ZSH_AI_COMMANDS_GPT_USER="Description of what the command should do:\n'''\n$ZSH_AI_COMMANDS_USER_QUERY\n'''\nGive me the appropriate command."
   ZSH_AI_COMMANDS_GPT_REQUEST_BODY='{
-    "model": "gpt-4-turbo-preview",
+    "model": "'$ZSH_AI_COMMANDS_LLM_NAME'",
+    "n": '$ZSH_AI_COMMANDS_N_GENERATIONS',
+    "temperature": 1,
     "messages": [
+      {
+        "role": "system",
+        "content": "'$ZSH_AI_COMMANDS_GPT_SYSTEM'"
+      },
       {
         "role": "user",
-        "content": "'$ZSH_AI_COMMANDS_GPT_MESSAGE_CONTENT'"
+        "content": "'$ZSH_AI_COMMANDS_GPT_USER'"
       }
     ]
   }'
 
-  ZSH_AI_COMMANDS_GTP_RESPONSE=$(curl -q --silent https://api.openai.com/v1/chat/completions \
+
+  fi
+
+
+
+  ZSH_AI_COMMANDS_GPT_RESPONSE=$(curl -q --silent https://api.openai.com/v1/chat/completions \
     -H "Content-Type: application/json" \
     -H "Authorization: Bearer $ZSH_AI_COMMANDS_OPENAI_API_KEY" \
     -d "$ZSH_AI_COMMANDS_GPT_REQUEST_BODY")
   local ret=$?
 
-  # replace all newlines with spaces
-  ZSH_AI_COMMANDS_COMMAND_SELECTION=$(echo $ZSH_AI_COMMANDS_GTP_RESPONSE | tr -d '\n')
-  # get the first suggestion
-  ZSH_AI_COMMANDS_COMMAND_SELECTION=$(echo $ZSH_AI_COMMANDS_COMMAND_SELECTION | jq -r '.choices[0].message.content')
-  # remove the string ```json from the beginning of the string
-  ZSH_AI_COMMANDS_COMMAND_SELECTION=$(echo $ZSH_AI_COMMANDS_COMMAND_SELECTION | sed 's/```json//g')
-  # remove the string ``` from the end of the string
-  ZSH_AI_COMMANDS_COMMAND_SELECTION=$(echo $ZSH_AI_COMMANDS_COMMAND_SELECTION | sed 's/```//g')
-
-  # FIXME: The json array cannot be parsed if there are too many / unexpected special characters in the response
-  ZSH_AI_COMMANDS_SELECTED_COMMAND=$(echo $ZSH_AI_COMMANDS_COMMAND_SELECTION | jq -r '.[]' | fzf)
-  BUFFER=$ZSH_AI_COMMANDS_SELECTED_COMMAND
+  if [ $ZSH_AI_COMMANDS_EXPLAINER = true ]
+  then
+      ZSH_AI_COMMANDS_SUGGESTIONS=$(echo $ZSH_AI_COMMANDS_GPT_RESPONSE | jq -r '.choices[].message.content' | uniq | sort | awk -F ' *## ' '!seen[$1]++' -)
+
+else
+  ZSH_AI_COMMANDS_SUGGESTIONS=$(echo $ZSH_AI_COMMANDS_GPT_RESPONSE | jq -r '.choices[].message.content' | uniq)
+fi
+
+  ZSH_AI_COMMANDS_SELECTED=$(echo $ZSH_AI_COMMANDS_SUGGESTIONS | fzf)
```

Comment:
`fzf` currently opens in fullscreen mode which is pretty annoying. This fixes it so that it appears inline:

```suggestion
  ZSH_AI_COMMANDS_SELECTED=$(echo $ZSH_AI_COMMANDS_SUGGESTIONS | fzf --height=~100%)
```